### PR TITLE
Move to the v1beta1 Conditions implementation.

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

See here for more context on what's changed (no breaking changes): knative/pkg#361

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [:no_good_man:] Includes [tests](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md#principles) (if functionality changed/added)
- [:no_good_man:] Includes [docs](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md)
for more details._

cc @mattmoor

### CODE REVIEW

File: pkg/apis/pipeline/v1alpha1/pipelinerun_types.go [hunk 1]
```diff
@@ -113,16 +114,20 @@ type PipelineTrigger struct {
 
 // PipelineRunStatus defines the observed state of PipelineRun
 type PipelineRunStatus struct {
-	Conditions duckv1alpha1.Conditions `json:"conditions"`
+	duckv1beta1.Status `json:",inline"`
```
Comment (mattmoor) [inline pkg/apis/pipeline/v1alpha1/pipelinerun_types.go]: One thing this includes that I don't see here is `ObservedGeneration`.

For mutable types, this is important to understand whether the overall Status reflects the latest Spec.

Basically, if `metadata.generation == status.observedGeneration`, then I can believe the conditions.

Not for this change, but probably worth an issue.

Comment (vdemeester) [reply pkg/apis/pipeline/v1alpha1/pipelinerun_types.go]: ah I see, yeah we'll need to update our usage of `Condition` to reflect that then :angel: 
(adding this to my todo :upside_down_face: )

**Comment by bobcatfish** [comment]
Thanks for keeping everything up to date @vdemeester !! :tada: 

One thing that seems to be missing from the existing commit messages - and from the linked knative/pkg PR - is kind of a higher level explanation of what's going on here (i.e. what is Conditions, why did it change to v1beta1, why do we want this change, why do we have to make it - im assuming some kind of co-vendoring situation) :pray: 

Anyway tho if you want to stick with what you've got now I won't be too upset, the messages _do_ have sweet emojis ;)

/lgtm

I wasn't clear if you wanted to make that `ObservedGeneration` change that @mattmoor is suggesting so:

/hold

(p.s. @mattmoor any plans on moving this lib into its own repo outside of pkg? would be pretty sweeeeet)

/meow space

**Comment by vdemeester** [comment]
> Thanks for keeping everything up to date @vdemeester !! tada
> 
> One thing that seems to be missing from the existing commit messages - and from the linked knative/pkg PR - is kind of a higher level explanation of what's going on here (i.e. what is Conditions, why did it change to v1beta1, why do we want this change, why do we have to make it - im assuming some kind of co-vendoring situation) pray
> 
> Anyway tho if you want to stick with what you've got now I won't be too upset, the messages _do_ have sweet emojis ;)
> 
> /lgtm

Lol :joy: Right, so I wasn't sure about what to include as explaination there, but it's a mix of "keeping things up-to-date" and "assuming some kind of co-vendoring situation" in the near future :dancer: 

> I wasn't clear if you wanted to make that `ObservedGeneration` change that @mattmoor is suggesting so:
> 
> /hold

I was gonna fix that in a follow-up but I could do it too :sweat_smile:

**Comment by vdemeester** [comment]
ahah :sweat: I need to rebase…

**Comment by bobcatfish** [comment]
/lgtm

i wasn't sure if you wanted to merge this as-is or you wanted to address @mattmoor 's comment first so feel free to remove this if you want to go ahead as-is:

/hold

**Comment by vdemeester** [comment]
/hold cancel

Let's do that in follow-up :angel:

**Pull request merged**